apps/api/src/app/widgets/usecases/get-subscriber-preference/get-subscriber-preference.usecase.ts

```diff
@@ -0,0 +1,59 @@
+import { Injectable } from '@nestjs/common';
+import { MessageTemplateRepository, NotificationTemplateRepository, SubscriberPreferenceRepository } from '@novu/dal';
+import { GetSubscriberPreferenceCommand } from './get-subscriber-preference.command';
+import { IPreferenceChannels } from '@novu/shared';
+import {
+  GetSubscriberTemplatePreference,
+  GetSubscriberTemplatePreferenceCommand,
+} from '../get-subscriber-template-preference';
+
+@Injectable()
+export class GetSubscriberPreference {
+  constructor(
+    private subscriberPreferenceRepository: SubscriberPreferenceRepository,
+    private notificationTemplateRepository: NotificationTemplateRepository,
+    private messageTemplateRepository: MessageTemplateRepository,
+    private getSubscriberTemplatePreferenceUsecase: GetSubscriberTemplatePreference
+  ) {}
+
+  async execute(command: GetSubscriberPreferenceCommand): Promise<ISubscriberPreferenceResponse[]> {
+    const templateList = await this.notificationTemplateRepository.getActiveList(
+      command.organizationId,
+      command.environmentId,
+      true
+    );
+
+    const result: ISubscriberPreferenceResponse[] = [];
+
+    for (const template of templateList) {
+      {
+        const buildCommand = GetSubscriberTemplatePreferenceCommand.create({
+          organizationId: command.organizationId,
+          subscriberId: command.subscriberId,
+          environmentId: command.environmentId,
+          template,
+        });
+
+        const templateSubscriberPreference = await this.getSubscriberTemplatePreferenceUsecase.execute(buildCommand);
```

Comment:
this is row that makes me think that this could be bad performance wise when user have a lot of templates...

Reply:
@davidsoderberg Yes I totally see that you mean there, let me look into it and see if I can make it better :)